File: Python/Problem_1897.py
from typing import DefaultDict, List

# Problem - 1897: https://leetcode.com/problems/redistribute-characters-to-make-all-strings-equal/
# Python3 Problem!
class Solution:
    def makeEqual(self, words: List[str]) -> bool:
        # ch_frequency = [0] * 26
        ch_frequency = DefaultDict(int)

        for word in words:
            for ch in word:
                ch_frequency[ch] += 1

        for ch in ch_frequency:
            if ch_frequency[ch] % len(words):
                return False
            
        return True


if __name__=="__main__":
    sol = Solution()

    print(sol.makeEqual(["abc","aabc","bc"]))
    print(sol.makeEqual(["ab","a"]))

File: Cpp/Problem_1897.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

// Problem - 1897: https://leetcode.com/problems/redistribute-characters-to-make-all-strings-equal/
// C++ Solution!
class Solution {
public:
	bool makeEqual(vector<string>& words) {
		vector<int> ch_frequency(26, 0);

		for (string word : words) {
			for (char ch : word) {
				ch_frequency[ch - 'a']++;
			}
		}

		for (int i = 0; i < 26; i++) {
			if (ch_frequency[i] % words.size() != 0)
				return false;
		}

		return true;
	}
};

int main() {
	Solution sol;

	vector<string> input = { "abc", "aabc", "bc" };
	cout << sol.makeEqual(input) << endl;
	input = { "ab", "a" };
	cout << sol.makeEqual(input) << endl;

	return 0;
}